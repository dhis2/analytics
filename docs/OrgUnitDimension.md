# OrgUnitDimension Component

The OrgUnitDimension component is a part of the DHIS2 analytics package. It is used to select organisation units, levels, and groups.

## Props

- `roots`: An array of strings representing the roots of the organisation units.
- `selected`: An array of objects representing the selected organisation units. Each object should have an `id` and `name` property, and may optionally have a `path` property.
- `onSelect`: A function that is called when an organisation unit is selected.
- `hideGroupSelect`: A boolean that determines whether the group select should be hidden. Defaults to `false`.
- `hideLevelSelect`: A boolean that determines whether the level select should be hidden. Defaults to `false`.
- `hideUserOrgUnits`: A boolean that determines whether the user organisation units should be hidden. Defaults to `false`.
- `warning`: A string that represents a warning message.

## Usage
```javascript
import { OrgUnitDimension } from '@dhis2/analytics'

<OrgUnitDimension
    roots={['A0000001234']}
    selected={[{ id: 'B0000001234', name: 'Unit B' }]}
    onSelect={(selected) => console.log('Selected:', selected)}
    hideGroupSelect={false}
    hideLevelSelect={false}
    hideUserOrgUnits={false}
    warning="This is a warning message"
/>
```

