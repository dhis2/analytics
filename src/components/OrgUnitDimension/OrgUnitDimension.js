import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import CircularProgress from '@material-ui/core/CircularProgress'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import sortBy from 'lodash/sortBy'

import i18n from '@dhis2/d2-i18n'
import { OrgUnitSelector, userOrgUnits } from '@dhis2/d2-ui-org-unit-dialog'

import {
    apiFetchOrganisationUnitGroups,
    apiFetchOrganisationUnitLevels,
    apiFetchOrganisationUnits,
} from '../../api/organisationUnits'

import { ouIdHelper } from '../../modules/ouIdHelper'

import { DIMENSION_ID_ORGUNIT } from '../../modules/fixedDimensions'
import styles from './styles/OrgUnitDimension.style'
import { colors } from '../../modules/colors'

export const defaultState = {
    root: undefined,
    roots: undefined,
    // use "selected" property for cloning org units while user org unit(s) is (are) selected
    selected: [],
    ouLevels: [],
    ouGroups: [],
    showOrgUnitsTree: true,
}

class OrgUnitDimension extends Component {
    constructor(props) {
        super(props)

        this.state = defaultState
        this.userOrgUnitIds = userOrgUnits.map(ou => ou.id)

        this.loadOrgUnitTree(props.d2, props.displayNameProperty)
        this.loadOrgUnitGroups(props.d2, props.displayNameProperty)
        this.loadOrgUnitLevels(props.d2)
    }

    componentDidUpdate(prevProps) {
        const previousId = prevProps.current ? prevProps.current.id : null
        const currentId = this.props.current ? this.props.current.id : null

        // remount org units selector component to ensure
        // only selected org units are expanded
        if (previousId !== currentId) {
            this.hideOrgUnitsTree()

            setTimeout(this.showOrgUnitsTree, 0)
        }
    }

    showOrgUnitsTree = () => {
        this.setState({ showOrgUnitsTree: true })
    }

    hideOrgUnitsTree = () => {
        this.setState({ showOrgUnitsTree: false })
    }

    getUserOrgUnitsFromIds = ids => {
        return userOrgUnits.filter(ou => ids.includes(ou.id))
    }

    onLevelChange = event => {
        const levelIds = event.target.value.filter(id => !!id)

        this.props.onSelect({
            dimensionId: DIMENSION_ID_ORGUNIT,
            items: [
                ...this.props.ouItems.filter(
                    ou => !ouIdHelper.hasLevelPrefix(ou.id)
                ),
                ...levelIds.map(id => {
                    const levelOu = this.state.ouLevels.find(ou => ou.id === id)

                    return {
                        ...levelOu,
                        id: ouIdHelper.addLevelPrefix(levelOu.id),
                    }
                }),
            ],
        })
    }

    onGroupChange = event => {
        const groupIds = event.target.value.filter(id => !!id)

        this.props.onSelect({
            dimensionId: DIMENSION_ID_ORGUNIT,
            items: [
                ...this.props.ouItems.filter(
                    ou => !ouIdHelper.hasGroupPrefix(ou.id)
                ),
                ...groupIds.map(id => {
                    const groupOu = this.state.ouGroups.find(ou => ou.id === id)

                    return {
                        ...groupOu,
                        id: ouIdHelper.addGroupPrefix(id),
                    }
                }),
            ],
        })
    }

    onDeselectAllClick = () =>
        this.props.onDeselect({
            dimensionId: DIMENSION_ID_ORGUNIT,
            itemIdsToRemove: this.props.ouItems.map(ou => ou.id),
        })

    loadOrgUnitTree = (d2, displayNameProperty) => {
        apiFetchOrganisationUnits(d2, displayNameProperty)
            .then(rootLevel => rootLevel.toArray())
            .then(roots => {
                this.setState({
                    roots,
                    root: roots[0],
                })
            })
    }

    loadOrgUnitGroups = (d2, displayNameProperty) => {
        apiFetchOrganisationUnitGroups(d2, displayNameProperty).then(
            organisationUnitGroups =>
                this.setState({ ouGroups: organisationUnitGroups })
        )
    }

    loadOrgUnitLevels = d2 => {
        apiFetchOrganisationUnitLevels(d2).then(organisationUnitLevels =>
            this.setState({
                ouLevels: sortBy(organisationUnitLevels, ['level']),
            })
        )
    }

    handleOrgUnitClick = (event, orgUnit) => {
        const selected = this.props.ouItems

        if (selected.some(ou => ou.path === orgUnit.path)) {
            this.props.onDeselect({
                dimensionId: DIMENSION_ID_ORGUNIT,
                itemIdsToRemove: [orgUnit.id],
            })
        } else {
            this.props.onSelect({
                dimensionId: DIMENSION_ID_ORGUNIT,
                items: [
                    ...selected,
                    {
                        ...orgUnit,
                        name: orgUnit.name || orgUnit.displayName,
                    },
                ],
            })
        }
    }

    handleUserOrgUnitClick = (event, checked) => {
        if (checked) {
            if (!this.state.selected.length) {
                this.setState({
                    selected: this.props.ouItems.slice(),
                })
            }

            this.props.onSelect({
                dimensionId: DIMENSION_ID_ORGUNIT,
                items: [
                    ...this.props.ouItems.filter(ou =>
                        this.userOrgUnitIds.includes(ou.id)
                    ),
                    userOrgUnits.find(ou => ou.id === event.target.name),
                ],
            })
        } else {
            if (
                this.props.ouItems.length === 1 &&
                this.state.selected.length > 0
            ) {
                this.props.onSelect({
                    dimensionId: DIMENSION_ID_ORGUNIT,
                    items: this.state.selected,
                })
            } else {
                this.props.onDeselect({
                    dimensionId: DIMENSION_ID_ORGUNIT,
                    itemIdsToRemove: [event.target.name],
                })
            }
        }
    }

    handleMultipleOrgUnitsSelect = orgUnits => {
        const selected = this.props.ouItems

        this.props.onSelect({
            dimensionId: DIMENSION_ID_ORGUNIT,
            items: [
                ...selected,
                ...orgUnits.reduce((obj, ou) => {
                    // avoid duplicates when clicking "Select children" multiple times
                    if (!selected.find(i => i.id === ou.id)) {
                        obj.push({ ...ou, name: ou.name || ou.displayName })
                    }

                    return obj
                }, []),
            ],
        })
    }

    render = () => {
        const ids = this.props.ouItems.map(ou => ou.id)
        const selected = this.props.ouItems.filter(
            ou =>
                !this.userOrgUnitIds.includes(ou.id) &&
                !ouIdHelper.hasLevelPrefix(ou.id) &&
                !ouIdHelper.hasGroupPrefix(ou.id)
        )

        const userOrgUnits = this.getUserOrgUnitsFromIds(ids)
        const level = ids
            .filter(ouIdHelper.hasLevelPrefix)
            .map(ouIdHelper.removePrefix)
        const group = ids
            .filter(ouIdHelper.hasGroupPrefix)
            .map(ouIdHelper.removePrefix)

        return (
            <Fragment>
                <DialogTitle>{i18n.t('Organisation units')}</DialogTitle>
                <DialogContent style={styles.dialogContent}>
                    {this.state.root && this.state.showOrgUnitsTree && (
                        <OrgUnitSelector
                            d2={this.props.d2}
                            root={this.state.root}
                            roots={this.state.roots}
                            selected={selected}
                            userOrgUnits={userOrgUnits}
                            level={level}
                            group={group}
                            levelOptions={this.state.ouLevels}
                            groupOptions={this.state.ouGroups}
                            onLevelChange={this.onLevelChange}
                            onGroupChange={this.onGroupChange}
                            onDeselectAllClick={this.onDeselectAllClick}
                            handleUserOrgUnitClick={this.handleUserOrgUnitClick}
                            handleOrgUnitClick={this.handleOrgUnitClick}
                            handleMultipleOrgUnitsSelect={
                                this.handleMultipleOrgUnitsSelect
                            }
                            checkboxColor="secondary"
                            deselectAllTooltipFontColor={colors.black}
                            deselectAllTooltipBackgroundColor={colors.greyLight}
                            displayNameProperty={this.props.displayNameProperty}
                            isUserDataViewFallback={true}
                        />
                    )}
                    {!this.state.root && (
                        <CircularProgress style={styles.loader} />
                    )}
                </DialogContent>
            </Fragment>
        )
    }
}

OrgUnitDimension.propTypes = {
    d2: PropTypes.object,
    displayNameProperty: PropTypes.string,
    onSelect: PropTypes.func,
    onDeselect: PropTypes.func,
    ouItems: PropTypes.array,
    current: PropTypes.object,
}

export default OrgUnitDimension
