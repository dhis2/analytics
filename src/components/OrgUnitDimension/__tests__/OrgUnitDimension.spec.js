import { CustomDataProvider } from '@dhis2/app-runtime'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import OrgUnitDimension from '../OrgUnitDimension.js'

jest.mock('@dhis2-ui/organisation-unit-tree', () => {
    const lib = jest.requireActual('@dhis2-ui/organisation-unit-tree')

    return {
        ...lib,
        OrganisationUnitTree: () => <div>Org unit tree component mock</div>,
    }
})

describe('OrgUnitDimension', () => {
    const onSelect = jest.fn()

    const props = {
        roots: [],
        selected: [],
        onSelect: onSelect,
        hideGroupSelect: false,
        hideLevelSelect: false,
        hideUserOrgUnits: false,
        warning: '',
    }

    const renderOrgUnitDimension = (props) =>
        render(
            <CustomDataProvider
                data={{
                    organisationUnitLevels: {
                        organisationUnitLevels: [
                            {
                                name: 'Chiefdom',
                                level: 3,
                                id: 'tTUf91fCytl',
                                displayName: 'Chiefdom',
                            },
                            {
                                name: 'District',
                                level: 2,
                                id: 'wjP19dkFeIk',
                                displayName: 'District',
                            },
                            {
                                name: 'Facility',
                                level: 4,
                                id: 'm9lBJogzE95',
                                displayName: 'Facility',
                            },
                            {
                                name: 'National',
                                level: 1,
                                id: 'H1KlN4QIauv',
                                displayName: 'National',
                            },
                        ],
                    },
                    organisationUnitGroups: {
                        organisationUnitGroups: [
                            {
                                name: 'CHC',
                                id: 'CXw2yu5fodb',
                                displayName: 'CHC',
                            },
                            {
                                name: 'Chiefdom',
                                id: 'gzcv65VyaGq',
                                displayName: 'Chiefdom',
                            },
                            {
                                name: 'CHP',
                                id: 'uYxK4wmcPqA',
                                displayName: 'CHP',
                            },
                            {
                                name: 'Clinic',
                                id: 'RXL3lPSK8oG',
                                displayName: 'Clinic',
                            },
                            {
                                name: 'Country',
                                id: 'RpbiCJpIYEj',
                                displayName: 'Country',
                            },
                            {
                                name: 'District',
                                id: 'w1Atoz18PCL',
                                displayName: 'District',
                            },
                            {
                                name: 'Eastern Area',
                                id: 'nlX2VoouN63',
                                displayName: 'Eastern Area',
                            },
                            {
                                name: 'Hospital',
                                id: 'tDZVQ1WtwpA',
                                displayName: 'Hospital',
                            },
                            {
                                name: 'MCHP',
                                id: 'EYbopBOJWsW',
                                displayName: 'MCHP',
                            },
                            {
                                name: 'Mission',
                                id: 'w0gFTTmsUcF',
                                displayName: 'Mission',
                            },
                            {
                                name: 'NGO',
                                id: 'PVLOW4bCshG',
                                displayName: 'NGO',
                            },
                            {
                                name: 'Northern Area',
                                id: 'J40PpdN4Wkk',
                                displayName: 'Northern Area',
                            },
                            {
                                name: 'Private Clinic',
                                id: 'MAs88nJc9nL',
                                displayName: 'Private Clinic',
                            },
                            {
                                name: 'Public facilities',
                                id: 'oRVt7g429ZO',
                                displayName: 'Public facilities',
                            },
                            {
                                name: 'Rural',
                                id: 'GGghZsfu7qV',
                                displayName: 'Rural',
                            },
                            {
                                name: 'Southern Area',
                                id: 'jqBqIXoXpfy',
                                displayName: 'Southern Area',
                            },
                            {
                                name: 'Urban',
                                id: 'f25dqv3Y7Z0',
                                displayName: 'Urban',
                            },
                            {
                                name: 'Western Area',
                                id: 'b0EsAxm8Nge',
                                displayName: 'Western Area',
                            },
                        ],
                    },
                }}
            >
                <OrgUnitDimension {...props} />
            </CustomDataProvider>
        )

    beforeEach(() => onSelect.mockClear())

    test('OrgUnitDimension matches the snapshot', () => {
        const { container } = renderOrgUnitDimension(props)

        expect(container).toMatchSnapshot()
    })

    test('OrgUnitDimension calls onSelect when an organisation unit is selected', async () => {
        const user = userEvent.setup()

        renderOrgUnitDimension(props)

        await user.click(screen.getByText('User organisation unit'))

        expect(onSelect).toHaveBeenCalledWith({
            dimensionId: 'ou',
            items: [
                { id: 'USER_ORGUNIT', displayName: 'User organisation unit' },
            ],
        })
    })

    test('OrgUnitDimension calls onSelect with an empty array when selection is cleared', async () => {
        const user = userEvent.setup()

        renderOrgUnitDimension({
            ...props,
            // make some selection to enable the deselect all button
            selected: [{ id: 'USER_ORGUNIT_CHILDREN', name: 'User sub-units' }],
        })

        await user.click(
            screen.getByRole('button', {
                name: 'Deselect all',
            })
        )

        expect(onSelect).toHaveBeenLastCalledWith({
            dimensionId: 'ou',
            items: [],
        })
    })
})
