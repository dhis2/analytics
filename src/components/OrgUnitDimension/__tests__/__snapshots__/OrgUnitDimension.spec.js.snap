// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgUnitDimension OrgUnitDimension matches the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="userOrgUnitsWrapper"
    >
      <label
        class="jsx-1193297444 dense"
        data-test="dhis2-uicore-checkbox"
      >
        <input
          class="jsx-1193297444"
          type="checkbox"
          value=""
        />
        <div
          class="jsx-1193297444 icon dense"
        >
          <svg
            class="jsx-4069423049 jsx-3997907426 "
            viewBox="0 0 14 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="jsx-4069423049 jsx-3997907426 background"
              height="14"
              rx="2"
              width="14"
              x="0"
              y="0"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 border"
              d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 checkmark"
              d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            />
            <rect
              class="jsx-4069423049 jsx-3997907426 indeterminate"
              height="2"
              width="8"
              x="3"
              y="6"
            />
          </svg>
        </div>
        User organisation unit
      </label>
      <label
        class="jsx-1193297444 dense"
        data-test="dhis2-uicore-checkbox"
      >
        <input
          class="jsx-1193297444"
          type="checkbox"
          value=""
        />
        <div
          class="jsx-1193297444 icon dense"
        >
          <svg
            class="jsx-4069423049 jsx-3997907426 "
            viewBox="0 0 14 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="jsx-4069423049 jsx-3997907426 background"
              height="14"
              rx="2"
              width="14"
              x="0"
              y="0"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 border"
              d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 checkmark"
              d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            />
            <rect
              class="jsx-4069423049 jsx-3997907426 indeterminate"
              height="2"
              width="8"
              x="3"
              y="6"
            />
          </svg>
        </div>
        User sub-units
      </label>
      <label
        class="jsx-1193297444 dense"
        data-test="dhis2-uicore-checkbox"
      >
        <input
          class="jsx-1193297444"
          type="checkbox"
          value=""
        />
        <div
          class="jsx-1193297444 icon dense"
        >
          <svg
            class="jsx-4069423049 jsx-3997907426 "
            viewBox="0 0 14 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="jsx-4069423049 jsx-3997907426 background"
              height="14"
              rx="2"
              width="14"
              x="0"
              y="0"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 border"
              d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            />
            <path
              class="jsx-4069423049 jsx-3997907426 checkmark"
              d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            />
            <rect
              class="jsx-4069423049 jsx-3997907426 indeterminate"
              height="2"
              width="8"
              x="3"
              y="6"
            />
          </svg>
        </div>
        User sub-x2-units
      </label>
    </div>
    <div
      class="orgUnitTreeWrapper"
    >
      <div>
        Org unit tree component mock
      </div>
    </div>
    <div
      class="selectsWrapper"
    >
      <div
        class="jsx-114080822 root"
        data-test="org-unit-level-select"
      >
        <div
          class="jsx-114080822 root-input"
        >
          <div
            data-test="dhis2-uicore-select"
          >
            <div
              class="jsx-2991488683 root dense"
              data-test="dhis2-uicore-select-input"
              tabindex="0"
            >
              <div
                class="jsx-2991488683 root-children"
              >
                <div
                  class="jsx-2370825775 jsx-4123978364 root"
                >
                  <div
                    class="jsx-1271387003 "
                    data-test="org-unit-level-select-placeholder"
                  >
                    Select a level
                  </div>
                </div>
              </div>
              <div
                class="jsx-2991488683 root-right"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.146 6.146a.5.5 0 01.638-.057l.07.057L8 9.293l3.146-3.147a.5.5 0 01.638-.057l.07.057a.5.5 0 01.057.638l-.057.07-3.5 3.5a.5.5 0 01-.638.057l-.07-.057-3.5-3.5a.5.5 0 010-.708z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="jsx-114080822 root"
        data-test="org-unit-group-select"
      >
        <div
          class="jsx-114080822 root-input"
        >
          <div
            data-test="dhis2-uicore-select"
          >
            <div
              class="jsx-2991488683 root dense"
              data-test="dhis2-uicore-select-input"
              tabindex="0"
            >
              <div
                class="jsx-2991488683 root-children"
              >
                <div
                  class="jsx-2370825775 jsx-4123978364 root"
                >
                  <div
                    class="jsx-1271387003 "
                    data-test="org-unit-group-select-placeholder"
                  >
                    Select a group
                  </div>
                </div>
              </div>
              <div
                class="jsx-2991488683 root-right"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.146 6.146a.5.5 0 01.638-.057l.07.057L8 9.293l3.146-3.147a.5.5 0 01.638-.057l.07.057a.5.5 0 01.057.638l-.057.07-3.5 3.5a.5.5 0 01-.638.057l-.07-.057-3.5-3.5a.5.5 0 010-.708z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="summaryWrapper"
    >
      <span
        class="summaryText"
      >
        Nothing selected
      </span>
      <div
        class="deselectButton"
      >
        <button
          class="jsx-3146657073 secondary small"
          data-test="dhis2-uicore-button"
          disabled=""
          type="button"
        >
          Deselect all
        </button>
      </div>
    </div>
    <style />
  </div>
</div>
`;
